src/ALE/MOM_ALE.F90

```diff
@@ -1051,13 +1077,32 @@ subroutine ALE_remap_velocities(CS, G, GV, h_old_u, h_old_v, h_new_u, h_new_v, u
   real :: h1(GV%ke)     ! A column of source grid layer thicknesses [H ~> m or kg m-2]
   real :: h2(GV%ke)     ! A column of target grid layer thicknesses [H ~> m or kg m-2]
   real :: h_neglect, h_neglect_edge  ! Tiny thicknesses used in remapping [H ~> m or kg m-2]
+  real :: rescale_coef  ! Factor that scales the baroclinic velocity to conserve ke [nondim]
+  real :: u_bt, v_bt    ! Depth-averaged velocity components [L T-1 ~> m s-1]
+  real :: ke_c_src, ke_c_tgt ! \int [u_c or v_c]^2 dz on src and tgt grids [H L T-1 ~> m2 s-1]
+  real, dimension(SZIB_(G),SZJ_(G)) :: du2h_tot  ! The rate of change of vertically integrated
```

Comment:
I agree that `du2h_tot` and `dv2h_tot` should have units of `[R Z L2 T-3 ~> W m-2]`, but as currently written they actually have units of `R H L2 T-3 ~> W m-2 or kg2 m-5 s-3]`, as is reflected by the consistent  `conversion` factors in the `register_diag_field` calls for these diagnostics on lines 365 and 371 above.

  This is easily corrected, though, if instead of using `GV%Rho0` where `du2h_tot` and `dv2h_tot` are calculated at lines 1167 and 1235, we were to use `GV%H_to_RZ`.  In Boussinesq mode, `GV%H_to_RZ = GV%Rho0*GV%H_to_m*US%m_to_Z`, but in non-Boussinesq mode the thicknesses already include the masses and `GV%H_to_RZ` is just the (constant) rescaling factor to convert them to units of `[kg m-2]`.

 This suggested change has the added benefit of not introducing the Boussinesq reference density (`GV%Rho0`) into non-Boussinesq calculations.   In non-Boussinesq mode, I have been working to avoid the use of `GV%H_to_m` or `GV%H_to_Z` whereever possible because they imply division by the arbitrary Boussinesq reference density.  In this case the various factors would cancel out mathematically, but multiplying and then diving by 1035 kg m-3 changes answers at roundoff, so they can not be cancelled out after people start using the code without changing answers.